File: codility/Ex7_CountBoundedSlices.js
function solution(K, A) {
    let n = A.length;
    let slices = 0;

    if (n < 1 || n > 100000) {
        console.log("Error: the size of A should be within the range [1..100,000");
        return;
    }

    let temp = null;
    let max = 0;
    let min = 0;

    for (let i = 0; i < n; i++) {
        max = A[i];
        min = A[i];
        for (let j = i; j < n; j++) {
            temp = A.slice(i, j + 1);
            if (A[j] > max) max = A[j];
            if (A[j] < min) min = A[j];
            // if ((Math.max(...temp) - Math.min(...temp)) <= K) {
            if ((max - min) <= K) {
                slices += 1;
            }
            else {
                break;
            }
        }

    } 

    if (slices > 1000000000) {
        return 1000000000;
    }

    return slices;
}

File: codility/Ex7_CountBoundedSlices.py
# First Trial 
# 60% (Timeout error with O(N^2))

def solution(K, A):
    cnt = len(A)
    
    for i in range(len(A)-1):
        temp_min = min(A[i], A[i+1])
        temp_max = max(A[i], A[i+1])
        if temp_max - temp_min > K:
            continue
        cnt += 1
        for j in range(i + 2, len(A)):
            temp_min = min(temp_min, A[j])
            temp_max = max(temp_max, A[j])
            if temp_max - temp_min <= K:
                cnt += 1
            else:
                break
    return cnt